src/Playwright/Helpers/Paths.cs

```diff
@@ -37,7 +37,14 @@ internal static string GetExecutablePath()
             if (!assemblyDirectory.Exists || !File.Exists(Path.Combine(assemblyDirectory.FullName, "Microsoft.Playwright.dll")))
```

Comment:
The second half of this condition appears to be unused? We check assemblyDirectory.Exists unconditionally on line 40.

Reply:
There's a (weird) condition that happens when the directory exists, but the file doesn't exist. We then _rewrite_ the directory from `Assembly.Location` which can be `null` for some cases (specifically, for single file applications). 

*BUT* for most cases, the that file will exist.